draft.md

```diff
@@ -84,14 +84,44 @@ The idempotency key that is supplied as part of every `POST` request MUST be uni
 
 How to make the key unique is up to the client and it's agreed protocol with the server. It is `RECOMMENDED` that `UUID` or a similar random identifier be used as the idempotency key.
 
+## Idempotency Key Validity and Expiry
 
-## Resource Behavior
+The server MAY enforce a time based idempotency keys, thus, be able to purge or delete a key upon its expiry. Resource SHOULD publish expiration policy related documentation.
 
-How an idempotency key is made unique key is determined by the client. We `RECOMMEND` using `UUID` or similar random string with enough entropy to avoid collisions.
+## Idempotency Fingerprint
```

Comment:
I like this section. But, we are introducing a new concept for the reader. This feels like writing a pattern vs writing an ID for an HTTP header. But let's keep it and see what other reviewers think about it.

Reply:
Yes i added this because we are talking about server behavior for using the key for a different payload, same payload etc.